apm/src/app/user/user.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UserService } from './user.service';
+
+@Component({
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+export class UserComponent implements OnInit {
+
+  public editForm!: FormGroup;
+  private firstName!: FormControl;
+  private lastName!: FormControl;
+  private readonly homepageUrl = "/welcome";
+  constructor(private readonly userService: UserService, private readonly router: Router) { }
+  
+  public get FirstName() {
```

Comment:
always specify the return types for getters and methods, that way you have typescript to indicate to you if you are returning something that is not expected from them